lib/page-object/element.js

```diff
@@ -1,24 +1,169 @@
+        .filter(hasValidValueIn(definition))
+        .join(', ')
+    );
   }
 
-  this.name = options.name;
-  this.selector = options.selector;
-  this.locateStrategy = options.locateStrategy || 'css selector';
+  this.selector = definition.selector;
+  this.locateStrategy = definition.locateStrategy || options.locateStrategy;
+  this.parent = options.parent;
 }
 
 Element.prototype.toString = function() {
-  return 'Element[name=@' + this.name + ']';
+  if (Array.isArray(this.selector)) {
+    return this.selector.join(',');
+  }
+  var prefix = this.constructor === Element ? '@' : '';
+  return this.constructor.name + '[name=' + prefix + getName(this) + ']';
+};
+
+/**
+ * Determines whether or not the element contians an @ element reference
+ * for its selector.
+ *
+ * @returns {boolean} True if the selector is an element reference starting with an
+ *    @ symbol, false if it does not.
+ */
+Element.prototype.hasElementSelector = function() {
+  return String(this.selector)[0] === '@';
+};
+
+/**
+ * If the element object requires a recursive lookup to resolve its
+ * selector, a new element containing the recursive lookup values
+ * is created and returned.
+ *
+ * @returns {Object} A new Element object containing the element and its
+ *    parents with a recursive lookup strategy for resolving the element
+ *    if one is needed. If not, null is returned.
+ */
+Element.prototype.getRecursiveLookupElement = function () {
+  var lookupList = getAncestorsWithElement(this);
+
+  if (lookupList.length > 1) {
+    return new Element({
+      selector: lookupList,
+      locateStrategy: 'recursion'
+    });
+  }
+
+  return null;
+};
+
+/**
+ * Copies selector properties to the first object from the second if the first
+ * object has undefined values for any of those properties.
+ *
+ * @param {Object} target The object to assign values to.
+ * @param {Object} source The object to capture values from.
+ */
+Element.copyDefaults = function(target, source) {
+  var props = ['name', 'parent', 'selector', 'locateStrategy'];
+  props.forEach(function(prop) {
+    if (target[prop] === undefined || target[prop] === null) {
+      target[prop] = source[prop];
+    }
+  });
+};
+
+/**
+ * Parses the value/selector parameter of an element command creating a
+ * new Element instance with the values it contains, if any. The standard
+ * format for this is a selector string, but additional, complex formats
+ * in the form of an array or object are also supported.
+ *
+ * @param {string|Object} value Selector value to parse into an Element.
+ * @param {string} [using] The using/locateStrategy to use if the selector
+ *    doesn't provide one of its own.
+ */
+Element.fromSelector = function(value, using) {
+
+  if (!value) {
+    throw new Error('Invalid selector value specified');
+  }
+
+  if (value instanceof Element) {
+    value.locateStrategy = value.locateStrategy || using;
+    return value;
+  }
+
+  if (using === 'recursion') {
```

Comment:
yeah, this looks good. Will still need to do a truthy check for value since typeof null = object, but this is still much more compact.